src/main/java/slimeknights/tconstruct/plugin/crt/managers/TinkerStationManager.java

```diff
@@ -0,0 +1,144 @@
+    addModifierRecipeInternal(name, inputs, toolRequired, modifierResult, modifierResultLevel, maxLevel, upgradeSlots, 0, modifierRequirements, minMatch, requirementsError);
+  }
+
+  @ZenCodeType.Method
+  public void addAbilityModifierRecipe(String name, IIngredientWithAmount[] inputs, IIngredient toolRequired, String modifierResult, int modifierResultLevel, int maxLevel, int abilitySlots, @ZenCodeType.Optional IData modifierRequirements, @ZenCodeType.OptionalInt(-1) int minMatch, @ZenCodeType.OptionalString(defaultError) String requirementsError) {
+    addModifierRecipeInternal(name, inputs, toolRequired, modifierResult, modifierResultLevel, maxLevel, 0, abilitySlots, modifierRequirements, minMatch, requirementsError);
+  }
+
+  @ZenCodeType.Method
+  public void addIncrementalModifierRecipe(String name, IIngredient input, int amountPerInput, int neededPerLevel, IIngredient toolRequirement, String modifierResult, int modifierResultLevel, int maxLevel, IItemStack leftover, @ZenCodeType.Optional IData modifierRequirements, @ZenCodeType.OptionalInt(-1) int minMatch, @ZenCodeType.OptionalString(defaultError) String requirementsError) {
+    addIncrementalModifierRecipeInternal(name, input, amountPerInput, neededPerLevel, toolRequirement, modifierResult, modifierResultLevel, maxLevel, 0, 0, leftover, modifierRequirements, minMatch, requirementsError);
+  }
+
+  @ZenCodeType.Method
+  public void addIncrementalUpgradeModifierRecipe(String name, IIngredient input, int amountPerInput, int neededPerLevel, IIngredient toolRequirement, String modifierResult, int modifierResultLevel, int maxLevel, int upgradeSlots, IItemStack leftover, @ZenCodeType.Optional IData modifierRequirements, @ZenCodeType.OptionalInt(-1) int minMatch, @ZenCodeType.OptionalString(defaultError) String requirementsError) {
+    addIncrementalModifierRecipeInternal(name, input, amountPerInput, neededPerLevel, toolRequirement, modifierResult, modifierResultLevel, maxLevel, upgradeSlots, 0, leftover, modifierRequirements, minMatch, requirementsError);
+  }
+
+  @ZenCodeType.Method
+  public void addIncrementalAbilityModifierRecipe(String name, IIngredient input, int amountPerInput, int neededPerLevel, IIngredient toolRequirement, String modifierResult, int modifierResultLevel, int maxLevel, int abilitySlots, IItemStack leftover, @ZenCodeType.Optional IData modifierRequirements, @ZenCodeType.OptionalInt(-1) int minMatch, @ZenCodeType.OptionalString(defaultError) String requirementsError) {
+    addIncrementalModifierRecipeInternal(name, input, amountPerInput, neededPerLevel, toolRequirement, modifierResult, modifierResultLevel, maxLevel, 0, abilitySlots, leftover, modifierRequirements, minMatch, requirementsError);
+  }
+
+  @Override
+  public void removeRecipe(IItemStack output) {
+    throw new IllegalArgumentException("Cannot remove Tinker Station Recipes by an IItemStack output! Use `removeByName(String name)` instead!");
```

Comment:
Removing by modifier ID might be useful In case I want to remove all recipes for haste for instance. Can do so using recipes that implement `IDisplayModifierRecipe`, assuming you can use an interface like that for removal.